app/models/hyrax/work.rb

```diff
@@ -7,38 +7,61 @@ module Hyrax
-  #   * Get admin set resource using: <code>admin_set = Hyrax.query_service.find_by(id: work.admin_set_id)</code>
-  #   * See 'a Hyrax::Work #admin_set_id' in /lib/hyrax/specs/shared_specs/hydra_works.rb
-  #     for tests of this relationship.
+  # * <b>Work to Administrative Set:</b> (1..1)  A work must be in one and only one admin set.
+  #   * Set admin set for a work using:
+  #       work.admin_set_id = admin_set.id
+  #   * Get admin set a work is in using:
+  #       admin_set = Hyrax.query_service.find_by(id: work.admin_set_id)
   #
   # Collection and Work
-  # * Collection to Work (0..m):  A collection can have many works.  This relationship
-  #   is defined by the inverse relationship stored in the work's attribute `:member_of_collection_ids`.
+  #
+  # * <b>Defined:</b> The relationship is defined by the work's `:member_of_collection_ids` attribute.
+  # * <b>Tested:</b> The relationship is tested in shared spec 'a Hyrax::Work' by testing
+  #   `it_behaves_like 'belongs to collections'`.  Shared specs are defined in /lib/hyrax/specs/shared_specs/hydra_works.rb.
+  # * <b>Collection to Work:</b> (0..m)  A collection can have many works.
   #   * See Hyrax::PcdmCollection for code to get works in a collection.
-  # * Work to Collection (0..m):  A work can be in many collections.  The
-  #   relationship to the collection is defined in the work's attribute `:member_of_collection_ids`.
+  # * <b>Work to Collection:</b> (0..m)  A work can be in many collections.
   #   * Add a work to a collection using Hyrax::CollectionMemberService (multiple method options)
-  #     * <code>Hyrax::CollectionMemberService.add_members(collection_id: col.id, members: works, user: current_user)</code>
-  #   * Get collection resources using: <code>collections = Hyrax.custom_queries.find_collections_for(resource: work)</code>
-  #   * See 'a Hyrax::Work' behaves_like 'belongs to collections' in
-  #     /lib/hyrax/specs/shared_specs/hydra_works.rb for tests of this relationship.
+  #       Hyrax::CollectionMemberService.add_members(collection_id: col.id, members: works, user: current_user)
+  #   * Get collections a work is in using:
+  #       collections = Hyrax.custom_queries.find_collections_for(resource: work)
   # @note Some collection types limit a work to belong to one and only one collection of that type.
   #
-  # Work and Work (TBD)
+  # Work and Work
+  #
+  # * <b>Defined:</b> The relationship is defined in the parent work's `:member_ids` attribute.
+  # * <b>Tested:</b> The relationship is tested in shared spec 'a Hyrax::Work' by testing
+  #   `it_behaves_like 'has_members'`.  Shared specs are defined in /lib/hyrax/specs/shared_specs/hydra_works.rb.
+  # * <b>Work to child Work:</b> (0..m)  A work can have many child works.
+  #   * Add a child work to a work using:
+  #       Hyrax::Transactions::Container['work_resource.add_to_parent']
+  #         .call(child_work, parent_id: parent_work.id, user: current_user)
```

Comment:
I'm not sure about recommending calling a transaction step outside the context of the full transaction.

### Potential Refactor Considerations

On the plus side, use of a transaction step, even outside of a full transaction, allows for the code to define what it means to add a child work to a work to be in only one place in the code.

Alternatives:  Collections have a service that adds/removes members of the collection.  Something similar could be added for works members.

Right now, the transaction to add a collection does not call the collection service.  And the collection service does not call the transaction.  It would be better if only one of those did the work and the other called to it.  I would tend to lean toward having the transaction call the service, but this is not a strong preference.

Interested in what others think.

If it's not a straight forward decision, I can break this out into an issue for further discussion.

Reply:
I was thinking the same and thought that this is what works now, but can be updated once there is a better pattern.
+1 to a separate issue for figuring out a better pattern.